### PR TITLE
learning operators with side predicates

### LINKED ISSUE #175: get RepeatedNextToEnv working with nsrt_learning
Currently, both planning and learning work only because we have set the threshold for being next to an object to be very low, such that whp the robot is only next to one object at any given time. Increasing this `nextto_thresh = 0.02` threshold would quickly break planning (and learning).

Requires some new ideas for planning or operator learning (or both).

### PR DESCRIPTION
**Pull request by tomsilver**
closes #175

### CODE REVIEW

File: src/nsrt_learning.py [hunk 1]
```diff
@@ -309,3 +436,136 @@ def unify_effects_and_options(
...
+                if ground_op.add_effects != segment.add_effects or \
+                   ground_op.delete_effects != segment.delete_effects:
+                    continue
+                # Operator covers the segment.
+                skeleton.append((ground_op.operator.name,
+                                 tuple(ground_op.objects)))
+                break
+            else:
+                continue
+            break
```
Comment (ronuchit) [inline src/nsrt_learning.py:483]: wtf

Comment (tomsilver) [reply src/nsrt_learning.py:483]: 😬

File: src/nsrt_learning.py [hunk 2]
```diff
@@ -146,9 +142,157 @@ def segment_trajectory(trajectory: GroundAtomTrajectory) -> List[Segment]:
...
+                print("Skeletons still covered; keeping conversion.")
+        # Consider converting each delete effect.
+        for effect in op.delete_effects:
+            if verbose:
+                print(f"Considering delete effect: {effect} from {op.name}")
+            current_op = name_to_strips_op[op.name]
+            assert current_op.name == op.name
+            # Tentatively replace the current operator.
+            remaining_params = {p for atom in current_op.preconditions | \
+                (current_op.add_effects - {effect}) | \
```
Comment (ronuchit) [inline src/nsrt_learning.py:228]: this was a bug right? this should be removed from delete_effects

Comment (tomsilver) [reply src/nsrt_learning.py:228]: yeah good catch

**Comment by ronuchit** [comment]
closing bc outdated, we can always revisit later in the closed PRs section

**Pull request closed without merging**